bcdaworker/worker/worker.go

```diff
@@ -342,14 +357,21 @@ func checkJobCompleteAndCleanup(ctx context.Context, r repository.Repository, jo
 			return false, err
 		}
 
-		if err := r.UpdateJobStatus(ctx, j.ID, models.JobStatusCompleted); err != nil {
-			return false, err
-		}
+		// make sure only inProgress jobs get moved to completed (don't update Cancelled jobs)
+		if j.Status != models.JobStatusCancelled {
```

Comment:
Can we move this conditional check outside of this for loop? So we can make the check after we've made the Completed check on line 329.

 I think logging and returning true is a good response.